README.md

```diff
@@ -195,6 +195,13 @@ More ready-to-use examples can be found in the [Example](https://mosecorg.github
   - `mosec_service_throughput` shows the service throughput.
 - Stop the service with `SIGINT` (`CTRL+C`) or `SIGTERM` (`kill {PID}`) since it has the graceful shutdown logic.
 
+## Performance tuning
+
+- Find out the best `max_batch_size` and `max_wait_time` for your inference service. The metrics will show the histograms of the real batch size and batch duration. Those are the key information to adjust these two parameters.
+- Try to split the whole inference process into different CPU and GPU stages (ref [DistilBERT](https://mosecorg.github.io/mosec/examples/pytorch.html#natural-language-processing)). The pipeline will make different stages run in parallel, which will fully utilize the GPU resource. Usually, higher GPU utilization will result in higher throughput. You can also adjust the number of workers in each stage to ensure that the GPU workers can get enough requests. Note that data passing through different stages will be serialized/deserialized by the `serialize_ipc/deserialize_ipc` methods, extremely large data might make the whole pipeline slower than before.
+- Use an appropriate `serialize/deserialize` method. By default, it's JSON. However, images and embeddings are not well supported by JSON. You can choose msgpack which is faster and binary compatible (ref [Stable Diffusion](https://mosecorg.github.io/mosec/examples/stable_diffusion.html)).
```

Comment:
```suggestion
- Note that data passing through different stages will be serialized/deserialized by the `serialize_ipc/deserialize_ipc` methods, so extremely large data might make the whole pipeline slower. You should choose an appropriate `serialize/deserialize` method. By default, it's JSON. However, images and embeddings are not well supported by JSON. You can choose msgpack which is faster and binary compatible (ref [Stable Diffusion](https://mosecorg.github.io/mosec/examples/stable_diffusion.html)).
```

Reply:
It should be split into two bullet points since IPC serde is different from serde. Usually people won't need to change the IPC serde, pickle is fast enough.

Reply:
Updated, PTAL.